lib/ansible/plugins/lookup/oc.py

```diff
@@ -0,0 +1,252 @@
+  set_fact:
+    project_fact:  "{{ lookup('oc',
+                       'Project',
+                       host=inventory_host,
+                       token=hostvars[inventory_host]['ansible_sa_token'],
+                       resource_name=project_name,
+                       validate_certs=validate_certs) }}"
+- name: Get All Service Accounts in a Project
+  set_fact:
+    service_fact: "{{ lookup('oc',
+                      'ServiceAccount',
+                      host=inventory_host,
+                      token=hostvars[inventory_host]['ansible_sa_token'],
+                      namespace=project_name,
+                      validate_certs=validate_certs) }}"
+"""
+
+RETURN = """
+  _list:
+    description:
+      - An object definition or list of objects definitions returned from OpenShift.
+    type: dict
+"""
+
+import json
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils import urls
+from ansible.module_utils.six.moves import urllib
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+
+
+class OC(object):
+    def __init__(self, host, port, token, validate_certs):
+        self.apis = ['api', 'oapi']
+        self.token = token
+        self.validate_certs = validate_certs
+        self.host = host
+        self.port = port
+        self.kinds = {}
+        bearer = "Bearer " + self.token
+        self.headers = {"Authorization": bearer}
+
+    def build_facts(self):
+
+        for api in self.apis:
+            url = "https://"
+            url += self.host
+            url += ":"
+            url += self.port
+            url += "/"
+            url += api
+            url += "/v1"
+            response = urls.open_url(url=url,
+                                     headers=self.headers,
+                                     validate_certs=self.validate_certs,
+                                     method='get')
+
+            if response.code >= 300:
+                raise AnsibleError("OC Query raised exception with code %s" +
+                                   "and message %s against url %s" %
+                                   str(response.code, response.read(), self.url()))
+
+            for resource in json.loads(response.read())['resources']:
+                if 'generated' not in resource['name']:
+                    self.kinds[resource['kind']] = \
+                        {'kind': resource['kind'],
+                         'name': resource['name'].split('/')[0],
+                         'namespaced': resource['namespaced'],
+                         'api': api,
+                         'version': 'v1',
+                         'baseurl': url
+                         }
+
+    def url(self, kind=None, namespace=None, resource_name=None, pretty=False, labelSelector=None, fieldSelector=None, resourceVersion=None):
+        firstParam = True
+        self.kind = kind
+
+        url = self.kinds[self.kind]['baseurl']
```

Comment:
In building up url in this function, it would be more pythonic to add each portion to a list and then use join() at the end to turn the list into a string.  Something like this:
``` python
url = [self.kinds[self.kind]['baseurl']]
        if self.kinds[self.kind]['namespaced'] is True:
            url.append('/namespaces/')
            if namespace is None:
                raise AnsibleError('Kind %s requires a namespace.  \
                                    None provided' % self.kind)
            url.append(namespace)
[...]
    return "".join(url)
```